apps/api/src/app/controllers/__tests__/edgeConnectionQueries.e2e.spec.ts

```diff
@@ -0,0 +1,254 @@
+import { INestApplication } from '@nestjs/common';
+import * as request from 'supertest';
+import { isDeepStrictEqual } from 'util';
+import { EdgeConnectionType } from '../../../domain/models/context/edge-connection.entity';
+import { isResourceCompositeIdentifier } from '../../../domain/models/types/entityCompositeIdentifier';
+import { resourceTypes } from '../../../domain/types/resourceTypes';
+import { InternalError } from '../../../lib/errors/InternalError';
+import { ArangoConnectionProvider } from '../../../persistence/database/arango-connection.provider';
+import { DatabaseProvider } from '../../../persistence/database/database.provider';
+import generateRandomTestDatabaseName from '../../../persistence/repositories/__tests__/generateRandomTestDatabaseName';
+import TestRepositoryProvider from '../../../persistence/repositories/__tests__/TestRepositoryProvider';
+import buildTestData from '../../../test-data/buildTestData';
+import formatResourceCompositeIdentifier from '../../../view-models/presentation/formatResourceCompositeIdentifier';
+import httpStatusCodes from '../../constants/httpStatusCodes';
+import createTestModule from './createTestModule';
+describe('When querying for edge connections', () => {
+    const testDatabaseName = generateRandomTestDatabaseName();
+
+    let app: INestApplication;
+
+    let arangoConnectionProvider: ArangoConnectionProvider;
+
+    let databaseProvider: DatabaseProvider;
+
+    let testRepositoryProvider: TestRepositoryProvider;
+
+    const { connections } = buildTestData();
+
+    beforeAll(async () => {
+        jest.resetModules();
+
+        const moduleRef = await createTestModule(testDatabaseName);
+
+        arangoConnectionProvider =
+            moduleRef.get<ArangoConnectionProvider>(ArangoConnectionProvider);
+
+        databaseProvider = new DatabaseProvider(arangoConnectionProvider);
+
+        testRepositoryProvider = new TestRepositoryProvider(databaseProvider);
+
+        app = moduleRef.createNestApplication();
+
+        await app.init();
+    });
+
+    beforeEach(async () => {
+        await testRepositoryProvider.testSetup();
+    });
+
+    afterEach(async () => {
+        await testRepositoryProvider.testTeardown();
+    });
+
+    describe(`GET /connections/`, () => {
+        it('should return the correct count', async () => {
+            const expectedResult = {
+                count: connections.length,
+            };
+
+            await testRepositoryProvider.getEdgeConnectionRepository().createMany(connections);
+
+            const result = await request(app.getHttpServer()).get(`/connections`);
+
+            expect(result.status).toBe(httpStatusCodes.ok);
+
+            expect(result.body).toEqual(expectedResult);
+        });
+    });
+
+    describe(`GET /connections/notes`, () => {
```

Comment:
We also need a test for the edge connection query endpoints and another for the resource endpoints that ensures errors are handled correctly when by chance there is a DTO that made it to the database although it fails invariant validation rules.